### PR TITLE
Add mysql support

### PR DESCRIPTION
**Pull request by yuval-siev**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1,10 +1,27 @@
...
+# POSTGRES Options:
+# DB_DRIVER="postgres"
+# # DB_POSTGRES_DSN = ""  # This overrides other postgres configurations
+# DB_POSTGRES_HOST="localhost"
+# DB_POSTGRES_USERNAME="postgres"
+# DB_POSTGRES_PASSWORD="postgres"
+# DB_POSTGRES_DB_NAME="postgres"
+# DB_POSTGRES_PORT=5432
+
+# SQLITE Options
```
Comment (drorIvry) [inline .env:16]: Mysql

File: .env [hunk 2]
```diff
@@ -1,10 +1,27 @@
...
+DB_DRIVER="mysql"
+# DB_MYSQL_DSN=""  # This overrides other mysqk configurations
+DB_MYSQL_USERNAME="root"
```
Comment (drorIvry) [inline .env:19]: because the name is literal to the DB I suggest reverting that back to a URL and driver type

File: dao/definitions_dao.go [hunk 3]
```diff
@@ -29,14 +29,15 @@ func initDefaultFields(taskDef *models.TaskDefinition) {
...
+	taskDef.NextExecutionTime = time.Now()
 
 	if taskDef.Namespace == "" {
 		taskDef.Namespace = "default"
 	}
 
-	if taskDef.ExecutionInterval > 0 {
-		taskDef.NextExecutionTime = time.Now().Add(time.Duration(taskDef.ExecutionInterval) * time.Second)
-	}
+	// if taskDef.ExecutionInterval > 0 {
```
Comment (drorIvry) [inline dao/definitions_dao.go:38]: remove if not needed

File: dao/definitions_dao.go [hunk 4]
```diff
@@ -79,14 +80,14 @@ func GetTaskDefinitionById(definitionId uuid.UUID) (models.TaskDefinition, error
...
-func UpdateDefinition(taskDefinition models.TaskDefinition) {
+func UpdateDefinition(taskDefinition *models.TaskDefinition) {
 	result := initializers.GetTaskDefinitionsTable().Where(
 		"id = ?",
 		taskDefinition.ID,
 	).Where(
 		"deleted = ?", // Can't update deleted definitions
 		false,
-	).Updates(
+	).Save(
```
Comment (drorIvry) [inline dao/definitions_dao.go:90]: hmm why this change?

File: initializers/database.go [hunk 5]
```diff
@@ -60,24 +64,77 @@ func connectSqlite() (*gorm.DB, error) {
...
-		config.DB_POSTGRES_DB_NAME,
-		config.DB_POSTGRES_PORT,
-		config.DB_POSTGRES_DSN_EXTRA,
-	)
+	var dsn string
+	if config.DB_POSTGRES_DSN != "" {
+		dsn = config.DB_POSTGRES_DSN
+	} else {
+		dsn = fmt.Sprintf(
+			"host=%s user=%s password=%s dbname=%s port=%d %s",
```
Comment (drorIvry) [inline initializers/database.go:72]: yeah it's more streamlined to just pass the uri

File: initializers/database.go [hunk 6]
```diff
@@ -60,24 +64,77 @@ func connectSqlite() (*gorm.DB, error) {
...
+		log.Error().Err(err).Str(
+			"table_name",
+			table_name,
+		).Msg(
+			"Could not AutoMigrate table",
+		)
+	}
+}
+
 func migrateTables() {
```
Comment (drorIvry) [inline initializers/database.go:127]: I Know it's not new but now we need to think about a single source of truth for the db schema, if it's go then we need to also use the CRUD in the UI

**Review by drorIvry — APPROVED** [review]

File: dao/definitions_dao.go (hunk 4 above)
Comment (yuval-siev) [reply dao/definitions_dao.go:90]: Updates doesn't update fields with 0 value.
So the `enabled=false` field wasn't updated. This meant that a one-time task would run again and again each time the poller runs

File: dao/definitions_dao.go (hunk 3 above)
Comment (yuval-siev) [reply dao/definitions_dao.go:38]: Done.

File: .env (hunk 1 above)
Comment (yuval-siev) [reply .env:16]: Fixed

File: .env (hunk 2 above)
Comment (yuval-siev) [reply .env:19]: Will do

File: initializers/database.go (hunk 5 above)
Comment (yuval-siev) [reply initializers/database.go]: Done.
